core/identity-spi/src/main/java/software/amazon/awssdk/identity/spi/TokenIdentity.java

```diff
@@ -33,4 +36,44 @@ public interface TokenIdentity extends Identity {
      * Retrieves string field representing the literal token string.
      */
     String token();
+
+    /**
+     * Constructs a new token object, which can be used to authorize requests to services that use token based auth
+     *
+     * @param token The token used to authorize requests.
+     */
+    static TokenIdentity create(String token) {
+        Validate.paramNotNull(token, "token");
+
+        return new TokenIdentity() {
```

Comment:
Can you refresh my mind why we're doing the identity as an inline definition? Is it because we already have a token elsewhere?

Reply:
Yeah, the token implementations we have are in other `modules` (sso/ssooidc). For AwsCredentials, there are implementations in the `auth` module. Used anonymous inner classes here to avoid creating a named classes like `AwsCredentialsIdentityImpl` or clashing names with `AwsBasicCredentials` and followed the same approach here. I went off a comment from Matt during the surface area review, but I might have misinterpreted it. LMK if you think this should be organized differently.

Reply:
I'd rather not create more classes right now, this seems easier to change if needed so I think we should keep it.